sqlmesh/core/snapshot/evaluator.py

```diff
@@ -985,6 +986,47 @@ def _get_adapter(self, gateway: t.Optional[str] = None) -> EngineAdapter:
             raise SQLMeshError(f"Gateway '{gateway}' not found in the available engine adapters.")
         return self.adapter
 
+    def _execute_virtual_statements(
+        self,
+        target_snapshots: t.Iterable[Snapshot],
+        snapshots: t.Dict[SnapshotId, Snapshot],
+        start: TimeLike,
+        end: TimeLike,
+        execution_time: TimeLike,
+        environment_naming_info: EnvironmentNamingInfo,
+        default_catalog: t.Optional[str] = None,
+        deployability_index: t.Optional[DeployabilityIndex] = None,
+    ) -> None:
+        """
+        Executes virtual statements for the provided target snapshots.
+        """
+
+        # Resolving the tables to their qualified view names.
+        table_mapping = to_view_mapping(
+            snapshots.values(),
+            environment_naming_info,
+            default_catalog=default_catalog,
+            dialect=self.adapter.dialect,
+        )
+
+        for snapshot in target_snapshots:
+            adapter = self._get_adapter(snapshot.model_gateway)
+            snapshot_deps = {snapshots[p_sid].name: snapshots[p_sid] for p_sid in snapshot.parents}
+            snapshot_deps[snapshot.name] = snapshot
```

Comment:
I wonder if we wanna make a helper for this, there are a few places where we do the same thing in the codebase:

- https://github.com/TobikoData/sqlmesh/blob/2adf5a1ba71deb6d0db63c8304f350b0a6d00c7e/sqlmesh/core/snapshot/evaluator.py#L722-L725
- https://github.com/TobikoData/sqlmesh/blob/2adf5a1ba71deb6d0db63c8304f350b0a6d00c7e/sqlmesh/core/snapshot/evaluator.py#L821-L824
- https://github.com/TobikoData/sqlmesh/blob/2adf5a1ba71deb6d0db63c8304f350b0a6d00c7e/sqlmesh/core/scheduler.py#L164-L167

cc @izeigerman